pkg/traces/traces.go

```diff
@@ -134,11 +136,17 @@ func newLogger(zapLevel zapcore.LevelEnabler, fmt logging.Format) *zap.Logger {
 		encoder = zaplogfmt.NewEncoder(config)
 	}
 
-	logger := zap.New(zapcore.NewCore(
-		encoder,
-		os.Stdout,
-		zapLevel,
-	), zap.AddCaller())
+	zapAdapter := zapadapter.New(l)
+
+	logger := zap.New(
+		zapcore.NewTee(
+			zapcore.NewCore(
+				encoder,
+				os.Stdout,
+				zapLevel,
+			),
+			zapAdapter.Core()),
+		zap.AddCaller())
```

Comment:
Is this necessary? It feels like we can use the following here to simplify:

```go
logger := zap.New(zapadapter.New(l))
```

(This is what we do for the otelcol Flow components)

Reply:
I believe we want to keep the previous definition of the traces logger as-is (utilising the customer encoder and level, plus the caller annotation), and just tee off to our own. WDYT?

Reply:
I personally would prefer removing the old encoder and force every log line to go through the same log.Logger instance that all our subsystems use. 

One of the big issues with _not_ using the log.Logger instance here is that changing the logging level at runtime doesn't change the verbosity of traces, since it only reads the desired logging level once. It also goes straight to os.Stdout, which means the Windows Event logs never include anything for tracing. 

Using zadapater exclusively will help us fix these issues IMO.

Reply:
This is done in https://github.com/grafana/agent/pull/2338/commits/3ebecda9fdddf48d6af8874148b4ec78dcdec261.

We can now use the server block to reload the logging level and format for traces, too.